ewebsock/src/native_tungstenite.rs

```diff
@@ -71,6 +71,8 @@ pub fn ws_receiver_blocking(url: &str, on_event: &EventHandler) -> Result<()> {
     let (mut socket, response) = match tungstenite::connect(url) {
         Ok(result) => result,
         Err(err) => {
+            let msg = format!("Connect: {err}");
+            on_event(WsEvent::Error(msg));
             return Err(err.to_string());
```

Comment:
This will effectively report the error twice - once to the callback, and once to the caller.

Can you explain why you need this? Returning the error is arguably better than the callback, since it means the error can be handed right away, and more ergonomic, and the `on_event` handler is only used for asynchronous events

Reply:
Then should this be moved to `ws_connect_impl` for native? The issue is that when a connection fails, the caller never gets notified of this error, since the error is just logged there rather than being passed back, that is why i thought of just adding it to the event handler: 

```
pub(crate) fn ws_connect_impl(url: String, on_event: EventHandler) -> Result<WsSender> {
    let (tx, rx) = std::sync::mpsc::channel();

    std::thread::Builder::new()
        .name("ewebsock".to_owned())
        .spawn(move || {
            if let Err(err) = ws_connect_blocking(&url, &on_event, &rx) {
                log::error!("WebSocket error: {err}. Connection closed."); // Logged error
            } else {
                log::debug!("WebSocket connection closed."); // OK message
            }
           // Join handle is of type (), so no error is passed back to the user
        })
        .map_err(|err| format!("Failed to spawn thread: {err}"))?;

    Ok(WsSender { tx: Some(tx) })
}
```

Reply:
Maybe an easier solution would be to call the on_event handler inside the connect_impl functions instead?

Reply:
Yes, calling `on_event` in instead of logging an error seems like the right choice to me, both in `ws_connect_impl` and `ws_receive_impl`